src/openms_gui/include/OpenMS/VISUAL/SpectraSelectionTabs.h

```diff
@@ -28,79 +28,82 @@
-// $Maintainer: Johannes Veit $
-// $Authors: Johannes Junker $
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
 // --------------------------------------------------------------------------
 
 #pragma once
 
 // OpenMS_GUI config
 #include <OpenMS/VISUAL/OpenMS_GUIConfig.h>
 
-//QT
-#include <QtWidgets/QTabBar>
-class QMouseEvent;
-class QMimeData;
+#include <OpenMS/KERNEL/StandardTypes.h>
+
+#include <QTabWidget>
 
 namespace OpenMS
 {
-  class String;
-
+  class SpectraViewWidget;
+  class SpectraIdentificationViewWidget;
+  class TOPPViewIdentificationViewBehavior;
+  class TOPPViewSpectraViewBehavior;
+  class TOPPViewBase;
   /**
-      @brief Convenience tab bar implementation
-
-      This tab bar differs in several ways from the QTabBar:
-      - you can close a tab by double-clicking it or through its context menu.
-      - it works based on tab identifiers (a fixed id stored in tab data) rather than on tab indices, which might
-        change when inserting or removing a tab.
-
-      @ingroup Visual
+    @brief A tabbed view, to browse lists of spectra or identifications
+    
   */
-  class OPENMS_GUI_DLLAPI TOPPASTabBar :
-    public QTabBar
+  class OPENMS_GUI_DLLAPI SpectraSelectionTabs
+    : public QTabWidget
   {
     Q_OBJECT
-public:
-    /// Constructor
-    TOPPASTabBar(QWidget * parent = nullptr);
-    /// Destructor
-    ~TOPPASTabBar() override;
-
-    /// Adds a new tab with the name @p text and the identifier @p id
-    int addTab(const String & text, int id);
-    /// Selects the tab with identifier @p id
-    void setCurrentId(int id);
-
-public slots:
-    /// Remove the tab with identifier @p id
-    void removeId(int id);
-
-signals:
-    /// Signal that indicates that the current tab changed
-    void currentIdChanged(int id);
-    /// Signal that indicates that the tab with identifier @p id is about to be removed (double click or context menu)
-    void aboutToCloseId(int id);
-
-    // /// Signal that is emitted, when a drag-and-drop action ends on a tab
-    // void dropOnTab(const QMimeData* data, QWidget* source, int id);
-    // /// Signal that is emitted, when a drag-and-drop action ends on the unused space on the right side of the tabs.
-    // void dropOnWidget(const QMimeData* data, QWidget* source);
-
-protected:
-    ///@name Reimplemented Qt events
+
+  public:
+    enum TAB_INDEX
+    {
+      SPECTRA_IDX = 0,  ///< first tab
+      IDENT_IDX = 1,    ///< second tab
+      AUTO_IDX          ///< automatically decide which tab to show (i.e. prefer IDENT_IDX if it has data)
+    };
+
+    /// Default constructor
+    SpectraSelectionTabs(QWidget* parent, TOPPViewBase* tv);
+
+    /// update items in the two tabs according to the currently selected layer
+    void update();
+
+    /// invoked when user changes the active tab to @p tab_index
+    void currentTabChanged(int tab_index);
+    
+    void showSpectrumAs1D(int index);
```

Comment:
if you know by chance what the next three methods are doing you could add a short comment

Reply:
done